queries/c/textobjects.scm

```diff
@@ -61,9 +61,15 @@
   (_) @statement.outer)
 
 ((parameter_list
-  (parameter_declaration) @parameter.inner . ","? @_end)
+  "," @_start . (parameter_declaration) @parameter.inner)
```

Comment:
You can write the `@parameter.inner` in another query, so we don't have duplicates and they also match functions with one parameter. Same for the other queries.

Reply:
This is possible, although I found another way to fix single parameter lists. If we do extract parameter.inner to a separate query then that would essentially be three queries for every kind of parameter. having 2 already feels unnecessary but I dont know how to do it more elegantly

Reply:
I think having three queries is more explicit, and we don't have to worry about always trying to capture the inner parameter, and that's one less query for treesitter.

Reply:
We always have to capture the @parameter.inner symbol as part of constructing the @parameter.outer range anwyay. (unless there is a way to do this more directly that I am not familiar with?)
It would just be `s/@parameter.inner/@_par` and adding a third query for the parameter inner. 

If there are performance implications it would be interesting to check, is there anyway to benchmark this?

Single parameter lists are fixed anyway

Reply:
>  It would just be s/@parameter.inner/@_par and adding a third query for the parameter inner.

Yeah, that's more explicit and would avoid bugs when changing those queries, like the one that was included.

> If there are performance implications it would be interesting to check, is there anyway to benchmark this?

Don't know how to benchmark, but I think these would just generate two matches for `parameter.inner` instead of just one.

Reply:
i see how it would be more explicit but I don't see how it would avoid bugs? if you wanted to change which nodes are parameters you would have to change 3 places most likely, (right now you have to change two places which is not great either)

Reply:
>  if you wanted to change which nodes are parameters you would have to change 3 places most likely

but you won't need to worry about changing or breaking `parameter.inner`, that one would always work, the other queries will only need to change `@_end` or `@_start` for `parameter.outer`